zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -230,29 +234,155 @@ fn setup_resolve_common_scalar_after_dispute<T: Config>(
+    let outcome = OutcomeReport::Categorical(0u16);
+    let _ = Call::<T>::report { market_id, outcome }
+        .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+
+    Ok((caller, market_id))
+}
+
 benchmarks! {
     admin_destroy_disputed_market{
-        let categories = T::MaxCategories::get();
-        let (caller, market_id) = setup_resolve_common_categorical::<T>(0, 0, categories)?;
+        let a in (T::MinCategories::get().into())..T::MaxCategories::get().into();
+        let d in 1..T::MaxDisputes::get();
+        let o in 0..63;
+        let c in 0..63;
+        let r in 0..63;
 
-        let categories : u32 = categories.saturated_into();
-        for i in 0..categories.min(T::MaxDisputes::get()) {
-            let origin = caller.clone();
-            let disputes = crate::Disputes::<T>::get(market_id);
-            let market = T::MarketCommons::market(&Default::default()).unwrap();
-            T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+        let (caller, market_id) = create_reported_market_with_pool::<T>(a)?;
+
+        let pool_id = T::MarketCommons::market_pool(&market_id).unwrap();
+
+        // minimum of number of disputes and (categories - 1) because of `matches_outcome_report`
+        for i in 1..=(a - 1).min(d) {
+            let outcome = OutcomeReport::Categorical(i.saturated_into());
+            let disputor = account("disputor", i, 0);
+            let dispute_bond = T::DisputeBond::get() +
+                (T::DisputeFactor::get() * i.saturated_into::<u32>().into());
+            T::AssetManager::deposit(Asset::Ztg, &disputor, dispute_bond).unwrap();
+            Pallet::<T>::dispute(RawOrigin::Signed(disputor).into(), market_id, outcome).unwrap();
+        }
+
+        let market = T::MarketCommons::market(&market_id.saturated_into()).unwrap();
+
+        let (range_start, range_end) = match market.period {
+            MarketPeriod::Timestamp(range) => (range.start, range.end),
+            _ => panic!("admin_destroy_reported_market: Unsupported market period"),
+        };
+
+        for i in 0..o {
+            MarketIdsPerOpenTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(range_start),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
+        for i in 0..c {
+            MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(range_end),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
+        let disputes = Disputes::<T>::get(market_id);
+        // TODO(#730): MarketIdsPerDisputeBlock will store the future block number.
+        let last_dispute = disputes.last().unwrap();
+        let dispute_at = last_dispute.at;
+        for i in 0..r {
+            MarketIdsPerDisputeBlock::<T>::try_mutate(
+                dispute_at,
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
```

Comment:
I am simulating the presence of other markets, yes. Because MarketIdsPerDisputeBlock has multiple ids inside it. I think the benchmark component comments for that are sufficient? 

 2c60636c8cee1f361935065554e836091bf03db5